mantidproject/mantid
# Indirect Diffraction - OSIRIS Diffspec calibration file option

Add option to use a calibration file for OSIRIS diffraction DiffSpec mode has been added in the interface and the underlying algorithm.

**To test:**

Tests need to be performed for the algorithm and the interface. 
- for both tests: **Ensure you turn on data archive searching and you will also need to have the `osiris_041_RES10` is available from the unit tests folder.**

**Algorithm**
- The following script should test the algorithm.

```
Diffraction_ws = ISISIndirectDiffractionReduction(InputFiles='OSIRIS00118782.nxs', CalFile='osiris_041_RES10.cal', Instrument='OSIRIS', SpectraRange='3,962')
```
- The output will be a workspace group, plot its member and ensure it looks like the diagram below

**Interface**
- Change the default facility/instrument to ISIS/OSIRIS (View > Preferences > Mantid)
- Open Indirect Diffraction (Interfaces > Indirect > Diffraction)
- Change the instrument to OSIRIS 
- Change diffraction to `Diffspec` mode.
- Ensure the default number of spectra are `3` to `962` 
- Use the run number 118782
- Ensure the `Cal File` option is greyed out initially
- Check the `Use Calibration File` option  
- Use the `osiris_041_RES10` calibration file.
- Click `Run`
- Plot the output workspace and ensure the plot is the following
  ![image](https://cloud.githubusercontent.com/assets/13132128/16555970/3f23d984-41cf-11e6-958c-c678ec736362.png)

Fixes #14794 

[RELEASE NOTES](https://github.com/mantidproject/mantid/blob/b119ee99a4a8b275d9bdf9bf0d7d230c8736ffe8/docs/source/release/v3.8.0/indirect_inelastic.rst)

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
